### PR TITLE
Configuration to specify no SSH key

### PR DESCRIPTION
**Pull request by austinmoore-**
Exposes configuration to allow a user to not specify an SSH key on an instance by setting `sshKeyName: ""`.

 [Docs](https://github.kdc.capitalone.com/identity-sre/kops/pull/4)

### CODE REVIEW

**Comment by chrisz100** [comment]
@austinmoore- could you please look into the rebase issue? Thank you!

**Comment by austinmoore-** [comment]
Rebased on master

File: upup/pkg/fi/cloudup/awstasks/sshkey.go [hunk 1]
```diff
@@ -196,6 +196,9 @@ func (e *SSHKey) IsExistingKey() bool {
 }
 
 func (e *SSHKey) TerraformLink() *terraform.Literal {
+	if e.NoSSHKey() {
```
Comment (rifelpet) [inline upup/pkg/fi/cloudup/awstasks/sshkey.go]: I think we'll need this same addition in `RenderCloudformation` a few lines below

File: protokube/pkg/gossip/mesh/mesh.pb.go [hunk 2]
```diff
@@ -30,18 +30,25 @@ limitations under the License.
...
-import strings "strings"
-import reflect "reflect"
-import github_com_gogo_protobuf_sortkeys "github.com/gogo/protobuf/sortkeys"
+	math "math"
 
-import io "io"
+	_ "github.com/gogo/protobuf/gogoproto"
+
+	errors "errors"
+
```
Comment (rdrgmnzs) [inline protokube/pkg/gossip/mesh/mesh.pb.go]: Any reason for the blank line between each import being added?

Comment (austinmoore-) [reply protokube/pkg/gossip/mesh/mesh.pb.go]: No reason. Not sure why it got formatted this way but I'll clean it up

**Comment by austinmoore-** [comment]
@rdrgmnzs How does it look?

**Comment by michaelajr** [comment]
Really hoping to get this merged for the next release. Any chance this will make it? @rdrgmnzs @chrisz100

**Comment by michaelajr** [comment]
@rdrgmnzs @chrisz100 @granular-ryanbonham hoping to get this merged for the next release. Can you (or someone) take a look?

**Comment by rifelpet** [comment]
@austinmoore- can you please rebase and fix the conflict? Once we get this up to date I'm hoping we can get it merged soon. thanks!

**Comment by austinmoore-** [comment]
@rifelpet Rebased

**Comment by austinmoore-** [comment]
/test pull-kops-e2e-kubernetes-aws

I think this might be a flake

**Comment by austinmoore-** [comment]
/test pull-kops-e2e-kubernetes-aws

Got different test failures this time so trying again

**Comment by austinmoore-** [comment]
/test pull-kops-e2e-kubernetes-aws

Jobs are finally passing, so lets try this again

**Comment by liuxunorg** [comment]
@austinmoore- We really need this feature.
Excuse me, Can you merged this feature into the [1.15](https://github.com/kubernetes/kops/tree/release-1.15) branch?

**Comment by rifelpet** [comment]
@austinmoore apologies for the continued delay. Do you think you could rebase this one final time? I can work on getting this reviewed and merged ASAP.

**Comment by austinmoore** [comment]
> @austinmoore apologies for the continued delay. Do you think you could rebase this one final time? I can work on getting this reviewed and merged ASAP.

I think you meant to write to @austinmoore- (see "-" on the end of his user name).

**Comment by rifelpet** [comment]
Oops, sorry about that :)

**Comment by rifelpet** [comment]
@austinmoore- looks great, thanks for sticking with it. If you can squash these commits then I think its ready to merge.

/approve

**Comment by rifelpet** [comment]
/lgtm
🎉

**Pull request merged**